### PR TITLE
codeinsights: Isolate oobmigration

### LINKED ISSUE #38051: RFC 697: Isolate out-of-band migrations (insights)
See [RFC 697: Multiple version upgrades](https://docs.google.com/document/d/1eNiwzWZmfpvzFhs9IDcM7Sbchv_Hm8Uj-6bI-3TDlvg).

Update out of band migration [#14](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@aed4b515082b4374f7d14fceeee98790c08a1361/-/blob/enterprise/internal/insights/migration/migration.go) so that it does not require code outside of a specific allow-list (which excludes things that transitively include `internal/conf`).

### LINKED ISSUE #40642: oobmigrations: Add tests for insights migration
The first draft of isolating code insights oobmigrations didn't have any unit tests to check validity. This PR tracks discovery and implementation of unit tests.

### PR DESCRIPTION
**Pull request by efritz**
**TL;DR**: Multi-version upgrades require us to keep the oobmigration _code_ around in-tree but frozen from external behavior changes, as we may have to run "old" code on instances coming up to speed. This PR moves the codeinsights oobmigration (recently deprecated) and removes all imports that are likely to change (excluding the stdlib and base db utils).

**Original message**: This PR *isolates* oobmigration [#14](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@aed4b515082b4374f7d14fceeee98790c08a1361/-/blob/enterprise/internal/insights/migration/migration.go) so that it does not need any _behavioral_ imports that can change over time. This is in preparation for its deprecation, where it will run only as part of the migrator for historic upgrades. We want to *freeze* this migration as much as possible, because we want to treat it with consistent behavior over time to allow the migrator to make some assumptions (like the registered code for an oobmigration is valid at any point in its lifetime).

Fixes #38051. Fixes #40642.

## Test plan

The current code in the PR has not been confidently tested yet. I plan to get a large data corpus from an old instance when insights values were still stored in site settings.

- [x] Add migration tests prior to merge

### CODE REVIEW

**Review by coury-clark — APPROVED** [review]
I'm basing a lot of my review on the generated output. It's a huge job and very difficult to check every detail, but what I have checked seems correct. Otherwise, logically, I don't see any problems here with isolating this job.

**Comment by efritz** [comment]
Merging now, but this migration will receive additional test coverage in #40700.
